File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int start = 0,end =nums.size();
        if(nums[0]<=nums[end-1]){
            return nums[0];
        }
        int m=INT_MAX;
        while(start<=end){
            int mid = (start+end)/2;
            m = min(m,nums[mid]);
            if(nums[mid]>=nums[0]){
                start = mid+1;
            }
            else{
                end = mid-1;
            }
        }
        return m;
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.ts
function findMin(nums: number[]): number {
    let start = 0, end = nums.length -1;
    let mid = 0;
    while(start <= end){
        mid = Math.floor((start + end) / 2);
        if(nums[mid]>nums[nums.length-1]){
            start = mid+1;
        }
        else{
            end = mid -1;
        }
    }
    return nums[start];
};